lab_2/main.py

```diff
@@ -141,4 +533,25 @@ def tokenize_big_file(path_to_file: str) -> tuple:
     :param path_to_file: a path
     :return: a tuple with ids
     """
-    pass
+    if not isinstance(path_to_file, str):
+        return ()
+
+    array = ()
+    tokens_dict = {}
+    for piece in tok(path_to_file):
+        for sentence_index in range(len(piece)):
+            array += piece[sentence_index]
+
+    for token in array:
+        if token not in cache:
+            if cache:
+                token_num = cache[list(cache.keys())[-1]] + 1
```

Comment:
Why is it `-1`?

Reply:
I assign a value based on the value of the previous key. This is necessary so that the values are the same for the same keys (tokens)